ibis/backends/duckdb/registry.py

```diff
@@ -0,0 +1,118 @@
+import sqlalchemy as sa
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.functions import GenericFunction
+
+import ibis.expr.operations as ops
+from ibis.backends.base.sql.alchemy import unary
+
+from ..postgres.registry import fixed_arity, operation_registry
+
+
+def _round(t, expr):
+    arg, digits = expr.op().args
+    sa_arg = t.translate(arg)
+
+    if digits is None:
+        return sa.func.round(sa_arg)
+
+    result = sa.func.round(sa_arg, t.translate(digits))
+    return result
+
+
+def _mod(t, expr):
+    left, right = map(t.translate, expr.op().args)
+    return left % right
+
+
+def _log(t, expr):
+    arg, base = expr.op().args
+    sa_arg = t.translate(arg)
+    if base is not None:
+        sa_base = t.translate(base)
+        if sa_base.value == 2:
+            return sa.func.log2(sa_arg)
+        elif sa_base.value == 10:
+            return sa.func.log(sa_arg)
+        else:
+            raise NotImplementedError
+        return sa.func.log(sa_base, sa_arg)
+    return sa.func.ln(sa_arg)
+
+
+class _regex_extract(GenericFunction):
+    def __init__(self, string, pattern, index):
+        super().__init__(string, pattern, index)
+        self.string = string
+        self.pattern = pattern
+        self.index = index
+
+
+@compiles(_regex_extract, 'duckdb')
+def _compile_regex_extract(element, compiler, **kw):
+    result = '(SELECT * FROM REGEXP_MATCHES({}, {}))[{}]'.format(
```

Comment:
Interesting. Is it necessary for this to be a scalar subquery? I would expect `REGEXP_MATCHES(...)[...]` to be sufficient.

Reply:
Ah, I don't think that's working currently -- I was fighting with some of the regexp stuff and trying things out.